### PR TITLE
Add onResourceStartForPlayer event, add getIsResourceRunningForPlayer function.

### PR DESCRIPTION
**Pull request by ghost**
This PR adds the function "getIsResourceRunningForPlayer", and the event "onResourceStartForPlayer".

```lua
bool getIsResourceRunningForPlayer(resource theResource, player thePlayer)
```

`onResourceStartForPlayer` is triggered on the started resource's root element, and the parameter is the player, which the resource has started for.

### CODE REVIEW

**Comment by TracerDS** [comment]
`onPlayerResourceStart` exists

As for the resource, `getResourceState` does the job for you. Simple triggerEvent to server if you need it server sided

File: Server/mods/deathmatch/logic/luadefs/CLuaResourceDefs.cpp [hunk 1]
```diff
@@ -708,6 +709,27 @@ int CLuaResourceDefs::getResourceState(lua_State* luaVM)
...
+        lua_pushboolean(luaVM, g_pGame->GetIsResourceRunningForPlayer(pResource, pPlayer));
+        return 1;
+    }
+    else
+        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
+
+    lua_pushboolean(luaVM, false);
+    return 1;
+}
+
```
Comment (TracerDS) [inline Server/mods/deathmatch/logic/luadefs/CLuaResourceDefs.cpp:732]: Use ArgumentParser

File: Server/mods/deathmatch/logic/CGame.cpp [hunk 2]
```diff
@@ -4754,3 +4768,29 @@ void CGame::ProcessClientTriggeredEventSpam()
...
+    {
+        auto& resources = entry.second;
+        resources.erase(std::remove(resources.begin(), resources.end(), pResource), resources.end());
+    }
+}
+
+bool CGame::GetIsResourceRunningForPlayer(CResource* pResource, CPlayer* pPlayer)
+{
+    auto it = m_mapClientStartedResources.find(pPlayer);
+    if (it != m_mapClientStartedResources.end())
```
Comment (TracerDS) [inline Server/mods/deathmatch/logic/CGame.cpp:4784]: Early return here

File: Server/mods/deathmatch/logic/CGame.cpp [hunk 3]
```diff
@@ -4146,6 +4154,12 @@ void CGame::Packet_PlayerResourceStart(CPlayerResourceStartPacket& Packet)
             CLuaArguments Arguments;
             Arguments.PushResource(pResource);
             pPlayer->CallEvent("onPlayerResourceStart", Arguments, NULL);
+
+            CLuaArguments Arguments2;
+            Arguments2.PushElement(pPlayer);
+            pResource->GetResourceRootElement()->CallEvent("onResourceStartForPlayer", Arguments2, NULL);
```
Comment (TracerDS) [inline Server/mods/deathmatch/logic/CGame.cpp:4160]: nullptr instead of NULL

File: Server/mods/deathmatch/logic/CGame.cpp [hunk 4]
```diff
@@ -1532,6 +1539,7 @@ void CGame::AddBuiltInEvents()
     // Resource events
     m_Events.AddEvent("onResourcePreStart", "resource", NULL, false);
     m_Events.AddEvent("onResourceStart", "resource", NULL, false);
+    m_Events.AddEvent("onResourceStartForPlayer", "player", NULL, false);
```
Comment (TracerDS) [inline Server/mods/deathmatch/logic/CGame.cpp:1542]: nullptr instead of NULL

**Review by TracerDS — CHANGES_REQUESTED** [review]

**Comment by DmitriyColeman** [comment]
onPlayerResourceStart already present.. 
You can create getIsResourceRunningForPlayer with already existed lua api

**Pull request closed without merging**